rosidl_gen/idl_generator.js

```diff
@@ -49,7 +54,95 @@ async function generateMessageJSStruct(messageInfo, dir) {
     spec: spec,
     json: JSON.stringify(spec, null, '  '),
   }));
-  await writeGeneratedCode(dir, fileName, generatedCode);
+  return writeGeneratedCode(dir, fileName, generatedCode);
+}
+
+async function generateActionJSStruct(actionInfo, dir) {
+  const spec = await parser.parseActionFile(actionInfo.pkgName, actionInfo.filePath);
+
+  const goalMsg = generateMessageJSStructFromSpec({
+    pkgName: actionInfo.pkgName,
+    subFolder: actionInfo.subFolder,
+    interfaceName: `${actionInfo.interfaceName}_Goal`
+  }, dir, spec.goal);
+
+  const resultMsg = generateMessageJSStructFromSpec({
+    pkgName: actionInfo.pkgName,
+    subFolder: actionInfo.subFolder,
+    interfaceName: `${actionInfo.interfaceName}_Result`
+  }, dir, spec.result);
+
+  const feedbackMsg = generateMessageJSStructFromSpec({
+    pkgName: actionInfo.pkgName,
+    subFolder: actionInfo.subFolder,
+    interfaceName: `${actionInfo.interfaceName}_Feedback`
+  }, dir, spec.feedback);
+
+  const sendGoalRequestSpec = actionMsgs.createSendGoalRequestSpec(actionInfo.pkgName, actionInfo.interfaceName);
```

Comment:
This is what I tried to do initially. It works for the response but not for the request because `<action name>_Goal` and the other suffixed messages do not exist on disk so the parsing fails.

Using parse_message_string we can only remove 1 of the 5 hardcoded specs so it's not really worth it.